packages/contentful/scripts/assign-users-as-lab-pi.ts

```diff
@@ -0,0 +1,196 @@
+  accessToken: contentfulManagementAccessToken,
+});
+
+const LIMIT = 1000;
+const queryOptions: QueryOptions = {
+  content_type: 'users',
+  'sys.archivedAt[exists]': false,
+  select: 'fields.oldLabs,fields.labs',
+  limit: LIMIT,
+};
+
+const fetchUsers = async (environment: Environment) => {
+  let userEntries: Entry[] = [];
+  const entries = await environment.getEntries(queryOptions);
+
+  userEntries.push(...entries.items);
+
+  const totalUsers = entries.total;
+  while (userEntries.length < totalUsers) {
+    let newEntries = await environment.getEntries({
+      ...queryOptions,
+      skip: userEntries.length,
+    });
+    userEntries.push(...newEntries.items);
+  }
+
+  return userEntries;
+};
+
+// Context:
+// users as a content type has a field called oldLabs which is a reference to labMembership entries
+// labMembership entries has a field called lab which is a reference to the lab entry
+// labs is a content type that has a field called name and another one call labPI which is a reference to a user entry
+// The goal is to assign the user as PI for the lab entry based on labMembership entries (which will be deprecated)
+// The user entry has a field called labs which is a reference to lab entries
+// write the script to update labs field with the lab entries that are referenced in oldLabs field through labMembership entries
+
+async function rollbackLabsMembership() {
+  try {
+    const space = await client.getSpace(spaceId);
+
+    const environment: Environment = await space.getEnvironment(environmentId);
+
+    // Fetch all users
+    const users = await fetchUsers(environment);
+
+    for (const user of users) {
+      if (
+        !user.fields ||
+        !user.fields.oldLabs ||
+        !user.fields.oldLabs['en-US'].length
+      ) {
+        console.log('No labs, skipping');
+        continue;
+      }
+
+      const labMemberships: Entry[] = user.fields.oldLabs?.['en-US'] || [];
+
+      if (!labMemberships.length) {
+        console.log('No labs, skipping - 2');
+        continue;
+      }
+
+      // Fetch labs referenced in `oldLabs`
+      const labs = await environment.getEntries({
+        content_type: 'labMembership',
+        'sys.id[in]': labMemberships
+          .map((labMembership) => labMembership.sys.id)
+          .join(','),
+      });
+
+      if (!labs.items.length) {
+        console.log(`No labs found for user: ${user.fields.email['en-US']}`);
+        continue;
+      }
+
+      const labIds = labs.items
+        .map((lab) => lab.fields.lab?.['en-US'].sys.id)
+        .filter(Boolean);
+
+      const isPublished = user.isPublished();
+
+      const userEntry = await environment.getEntry(user.sys.id);
+
+      // Update the labs field
+      userEntry.fields.labs = userEntry.fields.labs || {};
+      userEntry.fields.labs['en-US'] = labIds.map((labId) => ({
+        sys: {
+          type: 'Link',
+          linkType: 'Entry',
+          id: labId,
+        },
+      })) as Link<'Entry'>[];
```

Comment:
here it seems we are fetching the `labMemberships` just to get the `labIds`, right? but the `labIds` can be found on `labMemberships`, no? So couldn't we do something like:

```
const labMemberships: Entry[] = user.fields.oldLabs?.['en-US'] || [];

if (!labMemberships.length) {
  console.log('No labs, skipping - 2');
  continue;
}

const isPublished = user.isPublished();

const userEntry = await environment.getEntry(user.sys.id);

// Update the labs field
userEntry.fields.labs = userEntry.fields.labs || {};
userEntry.fields.labs['en-US'] = labMemberships.map((labMembership) => ({
  sys: {
    type: 'Link',
    linkType: 'Entry',
    id: labMembership.lab.sys.id,
  },
})) as Link<'Entry'>[];
```